File: BuildsArr/buildarr.go
package buildsarr

func buildArray(nums []int) []int {
	result := make([]int, len(nums))

	if len(nums) == 0 {
		return result
	}

	for i := 0; i < len(nums); i++ {
		result[i] = nums[nums[i]]
	}

	return result
}

File: BuildsArr/buildarr.js
/**
 * @param {number[]} nums - The array contain integer
 * @returns {number[]} returns the permutation array
 */
var buildArray = function (nums) {
	if (nums.length === 0) {
		return [];
	}

	const result = new Array(nums.length);

	for (let i = 0; i < nums.length; i++) {
		result[i] = nums[nums[i]];
	}

	return result;
};
